src/server/mod.rs

```diff
@@ -321,14 +321,23 @@ impl<State: Send + Sync + 'static> Server<State> {
 ///
 /// This type is useful only in conjunction with the [`HttpService`] trait,
 /// i.e. for hosting a Tide app within some custom HTTP server.
-#[derive(Clone)]
 #[allow(missing_debug_implementations)]
 pub struct Service<State> {
     router: Arc<Router<State>>,
     state: Arc<State>,
     middleware: Arc<Vec<Arc<dyn Middleware<State>>>>,
 }
 
+impl<State> Clone for Service<State> {
+    fn clone(&self) -> Self {
+        Self {
+            router: self.router.clone(),
+            state: self.state.clone(),
+            middleware: self.middleware.clone(),
+        }
+    }
+}
+
```

Comment:
What's the difference with the `Derive` block? It looks like there should be no difference?

Reply:
`derive` block adds `State: Clone` bound, which is undesirable.

Reply:
Oh I thought that bound was intentional. How else do we guarantee that the State can always be cloned for each route?

I feel like I'm missing something here 😅